sdk/python/kfp/compiler/v2_compat.py

```diff
@@ -126,13 +126,18 @@ def update_op(op: dsl.ContainerOp,
 
   for artifact_name, spec in sorted(
       component_spec.output_definitions.artifacts.items()):
+    metadata_path = op.file_outputs.get(artifact_name)
```

Comment:
This is rather hacky and should be temporary only. 
Maybe leave a TODO here and point to the issue? This should be reverted once we fix the sanitization.